### ISSUE TITLE
Example issues

### ISSUE THREAD

**Issue by mike2nl** [wemos/LOLIN_EPD_Library#2]
@wemos 
Hi guys and girls,

- in the LOLIN_EPD.h is the following defined:
EPD_BLACK,   ///< black color
EPD_WHITE,   ///< white color
EPD_INVERSE, ///< invert color
EPD_RED,     ///< red color
EPD_DARK,    ///< darker color
EPD_LIGHT,   ///< lighter color
The EPD_RED is used in the examples: testdrawrects(EPD_RED);
It is not working at all.

- The only things that is working is Black and White.
Why is this happen? It is a Waveshare display so far i know.

- The call: EPD.SetTextSize(); with the value 1...4 is not working at all.
What the hell is going on because the GFX library from Adafruit is used? ;-)

- Have i payed for something that can only show black/white/invers?

- Tested with the Wemos/Lolin ..
D1 Mini v1.0, v2.0, v3.1
D1 Mini Pro v2.0
ePaper 2.13 Shield v1.0.0

- Wemos TFT and I2C Connector Shield for D1 Mini is used too. 
https://github.com/wemos/D1_mini_Examples 
shows: add TFT I2C Connector Shield (V1.1.0 or later) support 5 months ago.
Which works with the 2.4 TFT Touch and the 1.4 TFT display (including address changes). 

So this is not the issue and the adresses are changed for the ePaper 2.13 Shield v1.0.0 too:
/*D1 mini*/
#define EPD_CS D4   // in place of D0
#define EPD_DC D8  // default

D0 is also not working for EPD_color_name. Only black/white/invers are working. 

PLease give some advise or more information. The Wemos WIki page tells nothing in that case.

**Comment by merdok** [comment]
I have the same issues. I am running the EPD shield with a D32 Pro and the examples with the RED color does not work correctly. Text color which is set to RED does not display at all.
Is this an issue or the display does not support RED at all? If the display does not support RED color then why it is included in the examples??

**Comment by mike2nl** [comment]
@merdok @wemos 
I think this is the data sheet and that one can black/white only...
Link: http://www.e-paper-display.cn/downloadRepository/4472ffa4-95d9-47fb-b0ee-7797d3b1cbeb.pdf

And here is the IL3897 conroller data sheet:
http://www.e-paper-display.com/downloadsfront.do?method=picker&flag=all&id=5f09fdc1-1054-40fc-b4f0-9f77579105b6&fileId=682&v=0.zip

It is that display type?
Link: http://www.e-paper-display.com/products_detail/productId=423.html

**Comment by merdok** [comment]
@mike2nl i think the first link is not the data sheet of the one used in the lolin EPD shield. The resolution does not match...
But it looks the same like on the product page and there only BLACK/WHITE support is specified. 

It is ok if the display would not support RED color but i am just wondering why do they do examples with RED color even if it is not supported??
I mean they clearly must have tested the code before they put it here...
@wemos can you explain this?

**Comment by mike2nl** [comment]
@merdok 
yes you are right about the data sheet. Got the wrong link. When you take that number on the flexible pcb from the display and search for it you get a different one.

And BTW: you will never get an answer from @wemos when it comes to mistakes or wrong product tests. I have tried it now for month and never got back an answer. Not via the offcial shop on Ali Express, or here or via mail. That´s typcial for Chinese companies.

I think it´s time to buy from western companies only, or i will start a new product line. I don´t have to make win on it, i need the costs back only. So...and we can suppport much newer sensors. Like the SHT31 or SHT35 in place of the SHT30. 
It´s all old stuff and no new things when it comes to sensors...

I will write my findings to the forum where i´m an admin and we have around 1.3 million followers
and they will take actions when i have written that test statement. Buy things from western companies.

Goob Bye Wemos...

**Comment by merdok** [comment]
@mike2nl 
Well that is a shame if they do not explain this...
And that is true about the sensors, they could offer shields with much newer sensors on them. This would be really great even if the cost would be higher.

**Comment by merdok** [comment]
@mike2nl 
Do you know if it is possible to do partial update on the display? It is really annoying when the full display is flashing black and white just to update a temperature value...

**Comment by mike2nl** [comment]
@merdok 
until now i had no time to do things with the display.
What i will do is to extract the display from the PCB with hot air. I have done that very often to get a part number. Possible i will find the original number and will chek to get a original display and test it.

So far as you can see we get no answer from @wemos.
I will make this now public in two very important foren.
1st one is Adafruit and the second one is Arduino. 
Will see which people had the same issue.
Next will be a lawyer who will look after that, so far we get no asnwer in 3 weeks from now on.

Because no mail answre via Ali Express, here and via one other channel.

**Comment by MrTanoshii** [comment]
I've been reading the docs, the IL3897 driver seems to only support mono color, so I'm guessing red is out of the equation.

Also I was looking at what waveshare sells on their store page

This seems to be the one we have from B/W wemos (note the resolution 250x122 and full refresh time 2s)
https://www.aliexpress.com/store/product/250x122-2-13inch-E-Ink-raw-display-panel-Without-PCB-Communicate-via-SPI-interface-Supports-various/216233_32813775523.html?spm=2114.12010612.8148356.8.2c261636U8IIbL

And this is the B/W/R (note resolution 212x104 and full refresh time 15s)
https://www.aliexpress.com/store/product/212x104-2-13inch-E-Ink-raw-display-panel-without-PCB-Red-black-white-Three-color-SPI/216233_32829493453.html?spm=2114.12010612.8148356.14.2c261636U8IIbL

--------------------------------------------------------------------------

I managed to get setTextSize() to work without any modifications, 

this is my fix for endPrintTest,
```
void endPrintTest() {
  EPD.clearBuffer();
  EPD.fillScreen(EPD_WHITE);
  EPD.setCursor(2, 0);
  EPD.setTextColor(EPD_BLACK);
  EPD.setTextSize(2);
  EPD.println("Hello World!");
  EPD.display(); // Can be omitted if #6 is committed
  EPD.setTextSize(1);
  EPD.setTextColor(EPD_BLACK);
  EPD.print(p, 6);
  EPD.println(" Want pi?");
  EPD.println(" ");
  EPD.print(8675309, HEX); // print 8,675,309 out in HEX!
  EPD.println(" Print HEX!");
  EPD.println(" ");
  EPD.setTextColor(EPD_BLACK);
  EPD.println("Sketch has been");
  EPD.println("running for: ");
  EPD.setTextColor(EPD_BLACK);
  EPD.print(millis() / 1000);
  EPD.setTextColor(EPD_BLACK);
  EPD.print(" seconds.");
  EPD.display();
}
```
For some reason it requires 2 EPD.display(), I also added the initial clearBuffer() and moved setCursor() to after fillScreen()

Note: #6 fixes the need for 2 EPD.display()
